ctapipe/instrument/InstrumentDescription.py

```diff
@@ -905,6 +907,222 @@ def load_config(filename):
+              np.array(yamlobject[item][data],\
+              names=yamlobject[item][header]))
+        elif ('Camera' or 'Cam') in item:
+            list_cameras[itemname] = table(\
+              np.array(yamlobject[item][data],\
+              names=yamlobject[item][header]))
+        elif ('Optics' or 'Opt') in item:
+            list_opticstypes[itemname] = table(\
+              np.array(yamlobject[item][data],\
+              names=yamlobject[item][header]))
+        elif ('Optics' or 'Opt') in item:
+            list_optics[itemname] = table(\
+              np.array(yamlobject[item][data],\
+              names=yamlobject[item][header]))
+        else:
+            list_auxiliaries[itemname] = table(\
+              np.array(yamlobject[item][data],\
+              names=yamlobject[item][header]))
+
+    # Another possibility: define inside pyyaml how to handle astropy.Quantity 
+    yaml.add_representer(Quantity, represent_quantity)
+
+
+    tel_table_prime = Table()
+   
+    for tel in list_telescopes:
+        # Basic type
+        telescope = list_telescopetypes[list_telescopes[tel]['BasedOn']]
+        # Change properties that only apply to this particular telescope
+        for item in list_telescopes[tel]['Override']:
+            telescope = override_in_table(telescope,item)
+    
+    for cam in list_cameras:
+        # Basic type
+        camera = list_cameratypes[list_cameras[cam]['BasedOn']]
+        # Change properties that only apply to this particular camera
+        for item in list_cameras[cam]['Override']:
+            camera = override_in_table(camera,item)
+    
+    for cam in list_optics:
+        # Basic type
+        optics = list_opticstypes[list_optics[cam]['BasedOn']]
+        # Change properties that only apply to this particular optics
+        for item in list_optics[cam]['Override']:
+            optics = override_in_table(optics,item)
+
+
+
+    for tel_id in list_telescopes:
+    try:
+        tel_table_prime['TelID'] = telescope[tel_id
+    except:
+
+
+    tel_posX = [random.uniform(1,100) for x in range(tel_num)]
+    tel_posY = [random.uniform(1,100) for x in range(tel_num)]
+    tel_posZ = [random.uniform(1,100) for x in range(tel_num)]
+    tel_table_prime['TelX'] = tel_posX
+    tel_table_prime['TelX'].unit = u.m
+    tel_table_prime['TelY'] = tel_posY
+    tel_table_prime['TelY'].unit = u.m
+    tel_table_prime['TelZ'] = tel_posZ
+    tel_table_prime['TelZ'].unit = u.m
+    mirror_area = [random.uniform(1,100) for x in range(tel_num)]
+    tel_table_prime['MirA'] = mirror_area
+    tel_table_prime['MirA'].unit = u.m**2
+    mirror_num = [random.randrange(1,124,1) for x in range(tel_num)]
+    tel_table_prime['MirN'] = mirror_num
+    foclen = [random.uniform(1,100) for x in range(tel_num)]
+    tel_table_prime['FL'] = foclen
+    tel_table_prime['FL'].unit = u.m
+    
+    telescope['TelescopeTable_Version%s' % version] = tel_table_prime    
+    
+    for t in range(len(tel_id)):       
```

Comment:
This is a python anti-pattern. Better is `for tel_id in tel_ids`. If you also need the index, use `for t, tel_id in enumerate(tel_ids)`